cookbook/ui_customization/create_a_new_filter_type.rst

```diff
@@ -0,0 +1,156 @@
+Create a custom product export builder filter
+=============================================
+
+When you create a new attribute type or if you want to change the rendering of a filter you need to create a custom filter. In this quick cookbook we we go throug all the step to acheive this task.
+
+Declare a filter provider for your custom filter:
++++++++++++++++++++++++++++++++++++++++++++++++++
+
+To create a custom filter, first we need to create a FilterProvider for our new filter type:
+
+.. code-block:: php
+    :linenos:
+    <?php
+
+    /*
+     * This file is part of the Akeneo PIM Enterprise Edition.
+     *
+     * (c) 2015 Akeneo SAS (http://www.akeneo.com)
+     *
+     * For the full copyright and license information, please view the LICENSE
+     * file that was distributed with this source code.
+     */
+
+    namespace Acme\Bundle\CustomBundle\Enrich\Provider\Filter;
+
+    use Pim\Bundle\EnrichBundle\Provider\Filter\FilterProviderInterface;
+    use Pim\Component\Catalog\Model\AttributeInterface;
+
+    class RangeFilterProvider implements FilterProviderInterface
+    {
+        /**
+         * {@inheritdoc}
+         */
+        public function getFilters($attribute)
+        {
+            return ['product-export-builder' => 'acme-range-filter'];
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function supports($element)
+        {
+            return $element instanceof AttributeInterface  &&
+                $element->getAttributeType() === 'pim_catalog_number' &&
+                null !== $element->getNumberMin() &&
+                null !== $element->getNumberMax();
+        }
+    }
+
+Now we can register this filter provider in our service.yml file
+
+.. code-block:: yaml
+    :linenos:
+
+    parameters:
+        acme.custom.provider.filter.range.class: Acme\Bundle\CustomBundle\Enrich\Provider\Filter\RangeFilterProvider
+
+    services:
+        acme.custom.provider.filter.range:
+            class: %acme.custom.provider.filter.range.class%
+            tags:
+                - { name: pim_enrich.provider.filter, priority: 110 }
+
+Your filter provider is now registered, congrats!
+
+Create the filter
++++++++++++++++++
+
+Now that we have a filter provider, we can create the filter itself:
+
+'use strict';
+/*
+ * src/Acme/Bundle/CustomBundle/Resources/public/js/product/filter/range-filter.js
+ */
+define([
+    'underscore',
+    'oro/translator',
+    'pim/filter/attribute/attribute',
+    'text!acme/template/product/filter/range',
+    'jquery.select2'
+], function (
+    _,
+    __,
+    AttributeFilter,
+    template
+) {
+    return AttributeFilter.extend({
```

Comment:
You should briefly explain your code. A block of code is not really interesting if it's not explained in a cookbook IMHO.